runtime/client_http_request.go

```diff
@@ -264,26 +264,35 @@ func (req *ClientHTTPRequest) executeDoWithRetry(ctx context.Context) (*http.Res
 			zap.Int("maxAttempts", req.timeoutAndRetryOptions.MaxAttempts),
 			zap.Bool("shouldRetry", shouldRetry))
 
-		//TODO (future releases) - make retry conditional, inspect error/response and then retry
+		// TODO (future releases) - make retry conditional, inspect error/response and then retry
 
-		//reassign body
+		// reassign body
 		if req.rawBody != nil && len(req.rawBody) > 0 {
 			req.httpReq.Body = io.NopCloser(bytes.NewBuffer(req.rawBody))
 		}
 
-		//Break loop if no retries
+		// Break loop if no retries
 		if !shouldRetry {
 			break
 		}
 	}
 	return nil, retryCount, err
 }
 
-//executeDo will send the request out with a timeout
+// executeDo will send the request out with a timeout
 func (req *ClientHTTPRequest) executeDo(ctx context.Context) (*http.Response, error) {
-	attemptCtx, cancelFn := context.WithTimeout(ctx, req.timeoutAndRetryOptions.RequestTimeoutPerAttemptInMs)
-	defer cancelFn()
-	return req.client.Client.Do(req.httpReq.WithContext(attemptCtx))
+	ctx, cancel := context.WithTimeout(ctx, req.timeoutAndRetryOptions.RequestTimeoutPerAttemptInMs)
+	defer cancel()
+	res, err := req.client.Client.Do(req.httpReq.WithContext(ctx))
+	// when no error, read body and capture before closing the connection
```

Comment:
nit: Add more context about context cancellation error if you read body outside the scope